embedded/document/engine.go

```diff
@@ -938,3 +938,79 @@ func (e *Engine) readMetadataAndValue(key []byte, atTx uint64, skipIntegrityChec
 
 	return entry.Metadata(), v, nil
 }
+
+func (e *Engine) DeleteDocument(ctx context.Context, collectionName string, query *protomodel.Query) (err error) {
+	sqlTx, err := e.sqlEngine.NewTx(ctx, sql.DefaultTxOptions())
+	if err != nil {
+		return mayTranslateError(err)
+	}
+	defer sqlTx.Cancel()
+
+	table, err := getTableForCollection(sqlTx, collectionName)
+	if err != nil {
+		return mayTranslateError(err)
```

Comment:
error translation should be responsability of the code directly using other packages, in this case it's a function defined in this package so it shouldn't be needed

Reply:
resolved